src/Http/WebUtilities/src/FileBufferingReadStream.cs

```diff
@@ -268,24 +270,34 @@ public override int Read(byte[] buffer, int offset, int count)
             return read;
         }
 
-        public override async Task<int> ReadAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
+        public override int Read(byte[] buffer, int offset, int count)
+        {
+            return Read(buffer.AsSpan(offset, count));
+        }
+
+        public override Task<int> ReadAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
+        {
+            return ReadAsync(buffer.AsMemory(offset, count), cancellationToken).AsTask();
+        }
+
+        public override async ValueTask<int> ReadAsync(Memory<byte> buffer, CancellationToken cancellationToken = default)
         {
             ThrowIfDisposed();
+
             if (_buffer.Position < _buffer.Length || _completelyBuffered)
```

Comment:
This logic isn't 100% right, we should be looping in here to fill buffer as much as we can before returning. That isn't a strict contract of Stream though...

Reply:
Should I wait for you to do this before reviewing, or do you want to merge this as is for now?

Reply:
I'm not making this change in the PR since it isn't a regression.